docs/source/mocks.rst

```diff
@@ -1,102 +1,73 @@
 
-How to use
------------------
+One powerful feature of a mock is that you can modify the value of variables within the smart contract.
+You could, for example, use this feature to test the behavior of a function that changes behavior depending on the value of a variable.
 
-.. container:: code-explanation
+Using mocks
+-----------
 
-  Mocks can be initialized only from deployed contracts. If we would have this contract:
+Initialization
+**************
 
-  .. code-block:: solidity
+Initialize with a contract name
+###############################
 
-    contract Counter {
-      uint256 public count;
+.. code-block:: typescript
 
-      constructor(uint256 _startAt) {
-        count = _startAt;
-      }
+  const myContractFactory = await smock.mock("MyContract");
+  const myContract = await myContractFactory.deploy(...);
 
-      function add(uint256 _amount) external {
-        count += _amount;
-      }
-    }
-
-.. container:: code-explanation
-
-  We could do something like this, combining real and faked logic:
-
-  .. code-block:: javascript
-
-    import { Counter, Counter__factory } from '@typechained';
-    import { MockContract, MockContractFactory, smock } from '@defi-wonderland/smock';
-
-    describe('Counter', () => {
-      let counterFactory: MockContractFactory<Counter__factory>;
-      let counter: MockContract<Counter>;
-
-      before(async () => {
-        counterFactory = await smock.mock('Counter');
-      });
-
-      beforeEach(async () => {
-        counter = await counterFactory.deploy();
-      });
-
-      it('...', async () => {
-        await counter.add(10);
-        expect(await counter.count()).to.equal(11);
-
-        await counter.count.returns(1);
-        expect(await counter.count()).to.equal(1);
-      });
-    });
+Using features of fakes
+***********************
 
+Mocks can use any feature available to fakes.
+See the documentation of `fakes <./fakes.html>`_ for more information.
 
 Call through
-------------
-
-.. container:: code-explanation
-
-  By default, mock functions will call the real contract logic.
-
-  .. code-block:: javascript
-
-    await mock.add(10);
-    await mock.count(); // returns 11
-
-    await mock.count.returns(1);
-    await mock.count(); // returns 1
+************
 
+Calls go through to contract by default
+#######################################
 
+.. code-block:: typescript
 
-Internal variables override
----------------------------
+  await myMock.add(10);
+  await myMock.count(); // returns 10
 
-.. container:: code-explanation
+  myMock.count.returns(1);
+  await myMock.count(); // returns 1
 
-  Set the value of an internal variable
+Manipulating variables
+**********************
 
-  .. code-block:: javascript
+Setting the value of a variable
+###############################
 
-    await mock.setVariable('_myInternalVariable', true);
+.. code-block:: typescript
 
-.. container:: code-explanation
+  await myMock.setVariable("myVariableName", 1234);
```

Comment:
Why the `"` instead of the `'`?

Reply:
Eh, just personal preference but happy to use `'` if desired.